### PR TITLE
[ADD] sale_website_cleanup: This module creates a cronjob to cancel o…

### PR DESCRIPTION
**Pull request by KarenKawaii**
…rders that have created more than 24 hours by the public user.
## [VX#5302](https://www.vauxoo.com/web#id=5302&view_type=form&model=project.task&menu_id=136&action=138)

In order to solve https://github.com/Vauxoo/instance/issues/244

### CODE REVIEW

File: sale_website_cleanup/README.rst [hunk 1]
```diff
@@ -0,0 +1,33 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Sale Website Cleanup
+====================
+
+This module add a cronjob that will canceled the orders that have created
```
Comment (nhomar) [inline sale_website_cleanup/README.rst]: s/canceled/cancel

Comment (nhomar) [inline sale_website_cleanup/README.rst]: s/that have created/that have been created/g

File: sale_website_cleanup/models/sales.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    _inherit = 'sale.order'
+
+    @api.multi
+    def cancel_old_orders(self):
+        sale_order_obj = self.env['sale.order']
+        rel_date = datetime.today() - relativedelta(days=1)
+        strf_date = rel_date.strftime('%Y-%m-%d %H:%M:%S')
+        draft_order = sale_order_obj.search([
+            ('state', '=', 'draft'),
+            ('partner_id.id', '=', self.env.ref('base.public_user')),
```
Comment (nhomar) [inline sale_website_cleanup/models/sales.py]: self.env.ref devuelve el browseable.

So.... el dominio es:

``` python
 ('partner_id', '=', self.env.ref('base.public_user'))`
```

Comment (nhomar) [inline sale_website_cleanup/models/sales.py]: As it fails, try:

```
 ('partner_id', '=', self.env.ref('base.public_user').id)
```

File: sale_website_cleanup/models/sales.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            ('date_order', '<=', strf_date)
+        ])
```
Comment (nhomar) [inline sale_website_cleanup/models/sales.py:32]: Include here a _logger.

```
_logger.info("Cancelling %s" % len(draft_order))
```

File: sale_website_cleanup/models/sales.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        for order in draft_order:
+            order.action_cancel()
```
Comment (nhomar) [inline sale_website_cleanup/models/sales.py:35]: @nhomar     Vauxoo member

Include here another logger _logger.

_logger.info("Cancelling this order id: %s" % order.id)

File: sale_website_cleanup/models/sales.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    @api.multi
+    def cancel_old_orders(self):
+        sale_order_obj = self.env['sale.order']
+        rel_date = datetime.today() - relativedelta(days=1)
+        strf_date = rel_date.strftime('%Y-%m-%d %H:%M:%S')
+        draft_order = sale_order_obj.search([
+            ('state', '=', 'draft'),
+            ('partner_id', '=', self.env.ref('base.public_user').id),
+            ('date_order', '<=', strf_date)
+        ])
```
Comment (nhomar) [inline sale_website_cleanup/models/sales.py:32]: Logger here. with the len of the orders.

File: sale_website_cleanup/models/sales.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        for order in draft_order:
+            order.action_cancel()
```
Comment (nhomar) [inline sale_website_cleanup/models/sales.py:35]: Logger here.

File: sale_website_cleanup/static/description/index.html [hunk 7]
```diff
@@ -0,0 +1,51 @@
+<head>
+  <style>
+      .backgrounds{background-color:#fff;color:#a41d35}
+  </style>
+</head>
+<section class="oe_container">
+ <div class="oe_row oe_spaced">
+  <p class="oe_mt32">
+   <strong>
+    This Addons add a cronjob that will canceled the orders that have created more than 24 hours.
```
Comment (nhomar) [inline sale_website_cleanup/static/description/index.html]: +This module add a cronjob that will cancel the orders that have been created
 +more than 24 hours.

**Comment by KarenKawaii** [comment]
Hi @nhomar could you check it please?

File: sale_website_cleanup/tests/test_sale_website_cleanup.py [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+# info Vauxoo (info@vauxoo.com)
+# coded by: karen@vauxoo.com
+# planned by: nhomar@vauxoo.com
+
+from openerp.tests.common import TransactionCase
+
+
+class TestSaleWebsiteCleanup(TransactionCase):
+    """Pseudo-constructor method of the setUp test."""
```
Comment (nhomar) [inline sale_website_cleanup/tests/test_sale_website_cleanup.py]: Incorrect docstring, explain what it is doing.

File: sale_website_cleanup/data/cancel_orders.xml [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+            <field name="active" eval="True"/>
+            <field name="user_id" ref="base.user_root"/>
+            <field name="interval_number">1</field>
+            <field name="interval_type">days</field>
+            <field name="numbercall">-1</field>
+            <field name="doall" eval="False"/>
+            <field name="model" eval="'sale.order'"/>
+            <field name="function" eval="'cancel_old_orders'"/>
+            <field name="args" eval="'(1,)'"/>
+            <field name="priority">5</field>
```
Comment (nhomar) [inline sale_website_cleanup/data/cancel_orders.xml]: put a higher priority 50 is Ok!

File: sale_website_cleanup/data/cancel_orders.xml [hunk 10]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+    <data noupdate="0">
+        <record id="ir_cron_cancel_orders" model="ir.cron">
+            <field name="name">Cancel orders that have created more than 24 hours</field>
+            <field name="active" eval="True"/>
+            <field name="user_id" ref="base.user_root"/>
+            <field name="interval_number">1</field>
+            <field name="interval_type">days</field>
```
Comment (nhomar) [inline sale_website_cleanup/data/cancel_orders.xml]: Weeks not days please.

**Pull request merged**